### PR TITLE
GitCommitBear: Test for duplicate shortlogs

### PR DESCRIPTION
**Pull request by meetmangukiya**
Fixes https://github.com/coala/coala-bears/issues/861

### CODE REVIEW

**Comment by meetmangukiya** [comment]
It'd be real nice if I can get some advice on how can I fix this tiny bit of missing coverage :disappointed: Those are just two lines.. :(

File: bears/vcs/git/GitCommitBear.py [hunk 1]
```diff
@@ -204,3 +213,17 @@ def check_body(self, body,
...
                                'Commit body lines should not exceed {} '
                                'characters.'.format(body_line_length))
+
+    def check_duplicate_shortlogs(self, shortlogs: list):
+        shortlogs_set = set(shortlogs)
+        if len(shortlogs_set) == len(shortlogs):
+            return False
+        else:
+            for log in shortlogs_set:
+                if log in shortlogs:
```
Comment (aptrishu) [inline bears/vcs/git/GitCommitBear.py]: `if log in shortlogs:`
why is this check necessary ?
A set can not have such elements which are not present in the parent list from which it is formed.

**Comment by Nosferatul** [comment]
@meetmangukiya  try to debug your tests using `pdb`, search on google for this.

**Comment by meetmangukiya** [comment]
> @meetmangukiya try to debug your tests using pdb, search on google for this.

@Nosferatul thanks :+1: seems interesting because I never used a debugger earlier :smile:

**Comment by meetmangukiya** [comment]
@underyx take a look, does it meet the requirement that you though of ? Do you think we need some kind of modification?

File: bears/vcs/git/GitCommitBear.py [hunk 2]
```diff
@@ -61,25 +61,31 @@ def run(self, allow_empty_commit_message: bool = False, **kwargs):
                                            allowed or not.
```
Comment (sils) [inline bears/vcs/git/GitCommitBear.py]: no option for this?

**Comment by sils** [comment]
this issue needs some design thinking. We can't just implement it like this.

File: bears/vcs/git/GitCommitBear.py [hunk 3]
```diff
@@ -61,25 +61,31 @@ def run(self, allow_empty_commit_message: bool = False, **kwargs):
...
         """
         with change_directory(self.get_config_dir() or os.getcwd()):
-            stdout, stderr = run_shell_command('git log -1 --pretty=%B')
-
-        if stderr:
-            self.err('git:', repr(stderr))
+            stdout_head, stderr_head = run_shell_command(
+                "git log -1 --pretty=%B")
+            stdout_shortlogs, stderr_shortlogs = run_shell_command(
+                "git log --pretty=%s")
```
Comment (meetmangukiya) [inline bears/vcs/git/GitCommitBear.py]: Oh, I assumed that just splitting at new lines wouldn't be a problem even if it were say 10k ...

I'll try to come up with something better

**Comment by aptrishu** [comment]
Marking it wip. You need to reinvent the idea.

File: bears/vcs/git/GitCommitBear.py [hunk 4]
```diff
@@ -64,7 +66,8 @@ def run(self, allow_empty_commit_message: bool = False, **kwargs):
             stdout_head, stderr_head = run_shell_command(
                 "git log -1 --pretty=%B")
             stdout_shortlogs, stderr_shortlogs = run_shell_command(
-                "git log --pretty=%s")
+                "git log -{} --pretty=%s".format(
```
Comment (jayvdb) [inline bears/vcs/git/GitCommitBear.py]: please use `--max-count=`

Comment (meetmangukiya) [inline bears/vcs/git/GitCommitBear.py]: done

**Comment by meetmangukiya** [comment]
`git coala` ? Passing `coala` as argument to `git`  ? :thinking:

**Comment by meetmangukiya** [comment]
And I guess that means close this PR? What about another related issue about checking commits passed on by CI, should that be closed too?

**Comment by jayvdb** [comment]
> git coala ? Passing coala as argument to git ?

As I understand it, interacting with multiple commits should be achieved by creating a tool which uses coala.  `git coala` would be one approach, and would work sort-of line [`git review`](https://github.com/openstack-infra/git-review), checking each commit before submitting it to be reviewed. And if someone doesnt use `git coala`, they will submit and any problems would be caught by `gitmate`.

I havent closed the PR, as you may want to discuss it further.

**Comment by meetmangukiya** [comment]
I thought that was like raising a PR at git to get coala command. But I guess we can mimic it without any permission. I think we should let gitmate handle this issue :+1:

**Pull request closed without merging**